pkg/secrets/vault.go

```diff
@@ -0,0 +1,167 @@
+package secrets
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/bruin-data/bruin/pkg/config"
+	"github.com/bruin-data/bruin/pkg/connection"
+	"github.com/bruin-data/bruin/pkg/logger"
+	"github.com/hashicorp/vault-client-go"
+	"github.com/hashicorp/vault-client-go/schema"
+	"github.com/pkg/errors"
+)
+
+func NewVaultClient(logger logger.Logger, host, token, role, path string, mountPath string) (*Client, error) {
+	if host == "" {
+		return nil, errors.New("no host provided")
```

Comment:
based on these I don't know what the env vars are, maybe do the validation in the run command already?